### PR TITLE
fix Logsoftmax infinity error

### LINKED ISSUE #1486: Sometimes training doesn't converge
We use latest git head 6602a8f6e8d7c550b363501e83fc20a28b8f6342 to do inception V1 and Resnet training, and find that training doesn't converge randomly:
# grep Loss bigdl.log | tail -n 10
2017-08-21 07:07:11 INFO  DistriOptimizer$:295 - [Epoch 82 184576/1281167][Iteration 58019][Wall Clock 251117.710963265s] Train 1792 in 4.220951966seconds. Throughput is 424.54877 records/second. Loss is NaN. Current learning rate is 0.022707158569988024. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:15 INFO  DistriOptimizer$:295 - [Epoch 82 186368/1281167][Iteration 58020][Wall Clock 251121.931915231s] Train 1792 in 4.147252403seconds. Throughput is 432.0933 records/second. Loss is NaN. Current learning rate is 0.022704307165623168. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:19 INFO  DistriOptimizer$:295 - [Epoch 82 188160/1281167][Iteration 58021][Wall Clock 251126.079167634s] Train 1792 in 4.264490471seconds. Throughput is 420.21432 records/second. Loss is NaN. Current learning rate is 0.022701455403109177. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:23 INFO  DistriOptimizer$:295 - [Epoch 82 189952/1281167][Iteration 58022][Wall Clock 251130.343658105s] Train 1792 in 4.155292104seconds. Throughput is 431.2573 records/second. Loss is NaN. Current learning rate is 0.02269860328231104. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:28 INFO  DistriOptimizer$:295 - [Epoch 82 191744/1281167][Iteration 58023][Wall Clock 251134.498950209s] Train 1792 in 4.492550308seconds. Throughput is 398.88257 records/second. Loss is NaN. Current learning rate is 0.022695750803093717. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:32 INFO  DistriOptimizer$:295 - [Epoch 82 193536/1281167][Iteration 58024][Wall Clock 251138.991500517s] Train 1792 in 4.197963127seconds. Throughput is 426.8737 records/second. Loss is NaN. Current learning rate is 0.022692897965322013. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:36 INFO  DistriOptimizer$:295 - [Epoch 82 195328/1281167][Iteration 58025][Wall Clock 251143.189463644s] Train 1792 in 4.15345278seconds. Throughput is 431.44824 records/second. Loss is NaN. Current learning rate is 0.022690044768860714. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:40 INFO  DistriOptimizer$:295 - [Epoch 82 197120/1281167][Iteration 58026][Wall Clock 251147.342916424s] Train 1792 in 4.074771818seconds. Throughput is 439.7792 records/second. Loss is NaN. Current learning rate is 0.02268719121357446. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:44 INFO  DistriOptimizer$:295 - [Epoch 82 198912/1281167][Iteration 58027][Wall Clock 251151.417688242s] Train 1792 in 3.758637876seconds. Throughput is 476.76846 records/second. Loss is NaN. Current learning rate is 0.02268433729932787. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:48 INFO  DistriOptimizer$:295 - [Epoch 82 200704/1281167][Iteration 58028][Wall Clock 251155.176326118s] Train 1792 in 4.106297326seconds. Throughput is 436.40286 records/second. Loss is NaN. Current learning rate is 0.02268148302598541. Current weight decay is 1.0E-4. Current momentum is 0.9.

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?

#1424 's optimization may cause an Infinity output when one of the input's bigger than 88.72283905206835.
Because `log(Float.MaxValue) == 88.72283905206835`.

This pr can fix this issue.

## How was this patch tested?

unit tests

## Related links or issues (optional)
fixed #1486

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LogSoftMax.scala [hunk 1]
```diff
@@ -77,37 +75,19 @@ class LogSoftMax[T: ClassTag](
...
-
-    ev.axpy(in.nElement,
-      sum,
-      ones,
-      0,
-      1,
-      out.storage.array,
-      out.storageOffset - 1,
-      1)
+    var logSum = ev.zero
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LogSoftMax.scala]: please use mkl to calculate this.

**Comment by qiuxin2012** [comment]
jenkins passed
